### PR TITLE
(MODULES-1387) Use ruby-bundler on ubuntu < 14.04

### PR DESCRIPTION
**Pull request by hunner**
And fix all the other failing tests

### CODE REVIEW

File: manifests/bundle.pp [hunk 1]
```diff
@@ -87,7 +87,7 @@
   }
 
   if $multicore {
-    if $multicore == '0' or $multicore >= $::processorcount {
+    if $multicore == '0' or versioncmp($multicore,$::processorcount) >= 0 {
```
Comment (hunner) [inline manifests/bundle.pp]: bastardization of a function due to the lack of decent string/integer comparisons.

File: manifests/params.pp [hunk 2]
```diff
@@ -17,7 +17,7 @@
       $rubygems_package = 'rubygems'
       $ruby_dev         = 'ruby-devel'
       $rubygems_update  = false
-      if $::operatingsystemmajrelease == 5 {
+      if $::operatingsystemmajrelease and versioncmp($::operatingsystemmajrelease, '5') == 0 {
```
Comment (bmjen) [inline manifests/params.pp]: Is this necessary? is there a circumstance where `$::operatingsystemmajrelease` would ever be undef/nil?

File: manifests/bundle.pp (hunk 1 above)
Comment (ghoneycutt) [inline manifests/bundle.pp]: To protect against multicore being specified as an integer, you could cast it as a string.

```puppet
if "${multicore}" == '0' or versioncmp("${multicore}",$::processorcount) >= 0 {
```

File: manifests/params.pp (hunk 2 above)
Comment (ghoneycutt) [inline manifests/params.pp]: Yes, on platforms that do not support it. I've certainly ran into this.

**Comment by ghoneycutt** [comment]
@hunner Sent you a PR to this branch which should fix the issue you're having with Travis.

https://github.com/hunner/puppetlabs-ruby/pull/1

File: .travis.yml [hunk 3]
```diff
@@ -6,11 +6,9 @@ script: "bundle exec rake validate && bundle exec rake lint && bundle exec rake
 matrix:
   fast_finish: true
   include:
-  - rvm: 1.8.7
-    env: PUPPET_GEM_VERSION="~> 3.0"
   - rvm: 1.9.3
     env: PUPPET_GEM_VERSION="~> 3.0"
-  - rvm: 2.0.0
```
Comment (ghoneycutt) [inline .travis.yml]: This would remove support for Ruby 1.8.7 and 2.0.0 though the README states that those are supported.

Comment (hunner) [inline .travis.yml]: Sigh. I don't want to support 1.8.7 any more. Ruby is EOL, puppet versions that only support that are eol, and PE versions using that are EOL. Are you saying I just need to delete it from the readme too? :)

Comment (ghoneycutt) [inline .travis.yml]: Well, I would strongly encourage you to keep support for 1.8.7 as it is what ships with EL6. Though if you want to remove it, I would also remove it from the README and do so in its own release that is separate from this PR, so it is clear that at release X, ruby 1.8.7 was dropped.

I don't personally use 2.0.0 and am not sure which OS's default to using it, though I have support for it in all of my modules. Would probably keep that around as it is newer.

**Pull request merged**